src/core/sock/sock-redirect.cpp

```diff
@@ -415,6 +418,97 @@ int init_child_process_for_nginx()
 
 #endif // DEFINED_NGINX
 
+#if defined(DEFINED_ENVOY)
+
+static int init_worker(int worker_id, int listen_fd)
+{
+    srdr_logdbg("worker: %d fd: %d", worker_id, listen_fd);
+
+    int ret = 0;
+    socket_fd_api *child_sock_fd_api = nullptr;
+
+    std::lock_guard<decltype(g_p_app->m_lock)> lock(g_p_app->m_lock);
+
+    if (worker_id > 0) {
+        sockinfo *si;
+        int parent_fd = -1;
+        auto itr1 = g_p_app->attr.envoy.map_thread_id.begin();
+        while (itr1 != g_p_app->attr.envoy.map_thread_id.end()) {
+            if (itr1->second == 0) {
+                auto itr2 = g_p_app->attr.envoy.map_listen_fd.begin();
+                while (itr1 != g_p_app->attr.envoy.map_listen_fd.end()) {
```

Comment:
itr1 is an iterator of map_thread_id but compared to map_listen_fd

Reply:
i guess these loops can be simplified with 
for (const auto& val1 : map_thread_id)
    for (const auto& val2 : map_listen_fd)